### PR TITLE
Only allow perf log control from the master app

### LINKED ISSUE #5377: Performance log will not be printed
If the master app has `print_perf_log = true` and sub app does not provide `Outputs` block. Should this be considered as an issue?

### PR DESCRIPTION
**Pull request by permcody**
closes #5377

Currently MOOSE only has a single perf_log. Putting in parameters to control it's output from subapps can conflict with options in the master. You can also easily get into scenarios where you end up with several copies of the perf_log.

I believe the best design with the current constraints is to simply not allow any perf_log control from the subapps.

### CODE REVIEW

File: test/tests/multiapps/full_solve_multiapp/master.i [hunk 1]
```diff
@@ -44,6 +44,7 @@
 
 [Outputs]
   exodus = true
+  print_perf_log = true
```
Comment (aeslaughter) [inline test/tests/multiapps/full_solve_multiapp/master.i]: Should there be a test spec to look for this warning?

Comment (permcody) [inline test/tests/multiapps/full_solve_multiapp/master.i]: Umm, yes, I'll add one

File: test/tests/misc/check_error/multi_master.i [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+[Outputs]
+  exodus = true
+  print_perf_log = true
+[]
+
+[MultiApps]
+  [./full_solve]
+    type = FullSolveMultiApp
+# not setting app_type to use the same app type of master, i.e. MooseTestApp
```
Comment (jwpeterson) [inline test/tests/misc/check_error/multi_master.i]: Comment is indented funny and should be on the first line of the block

File: framework/src/outputs/Console.C [hunk 3]
```diff
@@ -161,6 +161,22 @@ Console::Console(const InputParameters & parameters) :
     _setup_log = true;
   }
 
+  if (_app.name() != "main" &&
```
Comment (jwpeterson) [inline framework/src/outputs/Console.C]: This is quite an if statement!

**Review by jwpeterson — CHANGES_REQUESTED** [review]
Looks OK to me other than the one input file comment.

File: framework/src/outputs/Console.C (hunk 3 above)
Comment (permcody) [inline framework/src/outputs/Console.C]: It is... There are probably better ways of doing this but this works for now.

**Comment by friedmud** [comment]
This brings up a good point... how does the perf log work for MultiApps?  I guess you get all of the timing smashed together for all of the levels.  That's not very useful.

Is it time to undo having a "global" perf_log... and maybe move to one per app?

**Comment by permcody** [comment]
Yep - we need to allow perf logs for each app. I don't think it'll be a ton of work but it'll change the syntax of how we do logging everywhere since we'll move away from global to a member object.

I'll open a ticket for this work but I don't think it's a super high priority right now.

**Comment by permcody** [comment]
We'll have to be careful about how we log though. We don't really want to run the timer in one logger while we are running in another. It'll take some work to get this right.

**Comment by permcody** [comment]
@jwpeterson - comments are addressed and tests have passed.

**Review by jwpeterson — APPROVED** [review]

**Pull request merged**